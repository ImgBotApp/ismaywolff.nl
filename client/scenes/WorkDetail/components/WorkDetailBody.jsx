import React, { PropTypes } from 'react'
import format from 'date-fns/format'
import { gutter } from '../../../styles'
import { Title, SubTitle, TitleContainer } from '../../../components/title'
import { Cell, Grid } from '../../../components/grid'
import { ResponsiveZoomable } from '../../../components/images'

function WorkDetailBody({ work }) {
  return (
    <div>
      {/* Header */}
      <TitleContainer>
        <Title>{work.title}</Title>
        <SubTitle>{work.type} - {format(work.published, 'MMMM YYYY')}</SubTitle>
      </TitleContainer>

      {/* Body */}
      <p>{work.description}</p>

      {/* Gallery */}
      <Grid gutter={gutter}>
        { work.images.map(id => (
          <Cell
            gutter={gutter}
            smSize={1 / 1}
            mdSize={1 / 2}
            lgSize={1 / 3}
            key={id}
          >
            <ResponsiveZoomable id={id} />
          </Cell>
        ))}
      </Grid>
    </div>
  )
}

WorkDetailBody.propTypes = {
  work: PropTypes.object.isRequired
}

export default WorkDetailBody
