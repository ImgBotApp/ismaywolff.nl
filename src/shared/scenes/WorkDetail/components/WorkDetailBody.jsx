import React from 'react'
import { shape, string, arrayOf } from 'prop-types'
import dateformat from 'dateformat'
import { gutter } from '../../../styles'
import { DetailTitle } from '../../../components/title'
import { Cell, Grid } from '../../../components/grid'
import { Zoomable } from '../../../components/images'

function WorkDetailBody({ work }) {
  return (
    <div>
      {/* Header */}
      <DetailTitle
        main={work.title}
        sub={`${work.type} - ${dateformat(work.published, 'mmmm yyyy')}`}
      />

      {/* Body */}
      <p>{work.description}</p>

      {/* Gallery */}
      <Grid gutter={gutter}>
        {work.images.map(id => (
          <Cell gutter={gutter} smSize={1 / 1} mdSize={1 / 2} lgSize={1 / 3} key={id}>
            <Zoomable id={id} />
          </Cell>
        ))}
      </Grid>
    </div>
  )
}

WorkDetailBody.propTypes = {
  work: shape({
    title: string.isRequired,
    slug: string.isRequired,
    type: string.isRequired,
    published: string.isRequired,
    description: string.isRequired,
    images: arrayOf(string).isRequired,
    thumbnail: string.isRequired
  }).isRequired
}

export default WorkDetailBody
