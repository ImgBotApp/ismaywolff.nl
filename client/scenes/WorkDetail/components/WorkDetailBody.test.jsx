import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import WorkDetailBody from './WorkDetailBody'

describe('<WorkDetailBody />', () => {
  it('renders correctly', () => {
    const work = {
      title: 'title',
      type: 'type',
      published: 'published',
      description: 'description',
      images: ['one']
    }
    const wrapper = shallow(<WorkDetailBody work={work} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
