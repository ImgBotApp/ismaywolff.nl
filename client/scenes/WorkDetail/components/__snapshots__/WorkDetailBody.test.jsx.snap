// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkDetailBody /> renders correctly 1`] = `
<div>
  <glamorous(div)>
    <glamorous(h2)>
      title
    </glamorous(h2)>
    <glamorous(p)>
      type
       - 
      April 2011
    </glamorous(p)>
  </glamorous(div)>
  <p>
    description
  </p>
  <glamorous(div)
    gutter="0.5rem"
  >
    <glamorous(div)
      gutter="0.5rem"
      lgBreak="40em"
      lgSize={0.3333333333333333}
      mdBreak="25em"
      mdSize={0.5}
      smSize={1}
    >
      <ResponsiveZoomable
        id="one"
      />
    </glamorous(div)>
  </glamorous(div)>
</div>
`;
